frontend/src/book/store.js

```diff
@@ -131,16 +131,34 @@ export const setPageIndex = (pageIndex) => {
 };
 
 export const movePage = (delta) => {
+  if (delta > 0) {
+    animatePageTransition();
+  }
   setPageIndex(currentPage().index + delta);
 };
 
+const animatePageTransition = () => {
+  document.body.style.animation = 'bodyTransition 0.5s linear';
+  setTimeout(() => document.body.style.animation = '', 500);
+};
+
 export const moveStep = (delta) => {
   const stepIndex = currentStep().index + delta;
   const step = currentPage().steps[stepIndex];
   if (!step) {
     return;
   }
+  
   setUserStateAndDatabase(["pagesProgress", localState.user.pageSlug, "step_name"], step.name);
+  if (delta > 0) {
+    animateNextStep(stepIndex);
+  }
+};
+
+const animateNextStep = (stepIndex) => {
+  const div = document.getElementById(`step-text-${stepIndex}`);
+  div.style.animation = 'nextStepTransition 5s ease-out';
+  setTimeout(() => div.style.animation = '', 1000);
```

Comment:
This is close, but in terms of appearance:

1. It disappears quite abruptly, I think the setTimeout after 1 second is too short when the transition time is 5s. I played around a bit and liked 4s transition and 1.5s setTimeout. How's that seem to you?
2. There's a white gap above the text. Can you make the color fill the whole space between horizontal lines?

Reply:
It seems I was on another branch while testing these animations. I liked the 4s transition with 2s timeout. The <hr> element was causing the space to not be filled, so I had to change a bit how it is positioned in html. Looks better now, but do check it out.